### PR TITLE
Chore: Add @deprecation to LegacyForms

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Adds a deprecation comment, so that using one of these will be visible in the IDE
**Why do we need this feature?**
These were considered legacy components 3 years ago, so let's add people a reminder to migrate.

- Make people migrate
- Make people not import for new use cases

**Who is this feature for?**

Devs

**Which issue(s) does this PR fix?**:

Related to #76107, https://github.com/grafana/grafana/issues/22862 and https://github.com/grafana/grafana/issues/25841

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by ashharrison90 — APPROVED** [review]
nice 👍 

edit: do we have a plan for when we remove these entirely? 🤔

**Comment by tskarhed** [comment]
We can check how many imports they have in our plugin catalogue. If it is 0, I think we can remove it for next major version. Maybe @grafana/plugins-platform-frontend have a better idea.

File: packages/grafana-ui/src/components/Forms/Legacy/Input/Input.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import React, { PureComponent, ChangeEvent } from 'react';
 import { ValidationEvents, ValidationRule } from '../../../../types';
 import { validate, EventsWithValidation, hasValidationEvent } from '../../../../utils';
 
+/** @deprecated Please use the non legacy `Input` component, which does not require this enum. */
```
Comment (academo) [inline packages/grafana-ui/src/components/Forms/Legacy/Input/Input.tsx:7]: What about?

`Please use the `Input` component, which does not require this enum.` 

Better than using negatives.

**Review by academo — COMMENTED** [review]
nice! can you remove the `non-legacy` part from all the comments? as a developer I assume that the recommended component is already the current one and there's no need to tell it is the non-legacy

**Comment by tskarhed** [comment]
@academo I get that, but isn't it a bit confusing when the legacy component has the same name as the newer one?

**Comment by academo** [comment]
> @academo I get that, but isn't it a bit confusing when the legacy component has the same name as the newer one?

Maybe you can add a code snippet to clarify it?

**Comment by tskarhed** [comment]
@academo Do you think a link to the Storybook story would be helpful?

**Review by academo — APPROVED** [review]
Thanks for the improvements in the comments!

File: packages/grafana-ui/src/components/index.ts [hunk 2]
```diff
@@ -265,6 +265,7 @@ export { type UserView } from './UsersIndicator/types';
 export { InlineFormLabel } from './FormLabel/FormLabel';
 export { Divider } from './Divider/Divider';
 
+/** @deprecated Please use non-legacy versions of these components */
```
Comment (simPod) [inline packages/grafana-ui/src/components/index.ts:268]: Where are they?

Comment (Clarity-89) [reply packages/grafana-ui/src/components/index.ts:268]: If you go to the file for each of these components, you can see a more detailed deprecation message with the suggested alternatives.

**Pull request merged**